### DISCUSSION TITLE
Building KOReader in Mac OS: Error: Lua 5.1 interpreter not found at /opt/homebrew

### DISCUSSION THREAD

**Discussion by zhaokoh** [koreader/koreader#11711] [Q&A] [answered]
Hi,

I have just discovered KOReader and wondering where I have been in the last few years!!! The app is so well done and I really like the reflow + highlight functionality, which is exactly what I have been searching for decades!

I am trying to get KOReader synced and used in Mac OS based on the building instructions. However, I got this message: "Error: Lua 5.1 interpreter not found at /opt/homebrew."

I checked all the paths, and it seems okay. So, I think the problem is the Lua version. Lua 5.1 has been deprecated so when I did ```brew install lua@5.1```, it said:

```
(base) $ brew install lua@5.1
Error: lua@5.1 has been disabled because it is deprecated upstream!
```

Any suggestion? I am new to Lua and not sure if I can update the Lua version before building?

Thanks!

**Comment by benoit-pierre** [comment, marked as answer]
You can try the [workaround from #11660](https://github.com/koreader/koreader/pull/11660/files#diff-5c3fa597431eda03ac3339ae6bf7f05e1a50d6fc7333679ec38e21b337cb6721R41):
```bash
▸ brew fetch lua@5.1
▸ brew install "$(brew --cache lua@5.1)"
```

**Reply by zhaokoh** [reply]
Thank you so much @benoit-pierre, it works!

**Comment by Frenzie** [comment]
> I really like the reflow + highlight functionality

The reflow comes from <https://www.willus.com/k2pdfopt/>, which is definitely worth a look if you like reflow.

**Reply by zhaokoh** [reply]
Thanks @Frenzie! Yes, I tried to incorporate k2pdfopt into my reading workflow but can't find the optimal configuration. KOReader has the ability to highlight on a reflow document - that's amazing!
